### PR TITLE
[TRAFFIC-10636] Add `confluent network region list`

### PR DESCRIPTION
**Pull request by ay-zhang**
What
----

Note that this is merging into the feature branch https://github.com/confluentinc/cli/tree/traffic-8850-network-cli-az. We will merge commits into main from this feature branch after all of the commands for network are implemented.

- Add `confluent network region list`
- Autocomplete for `confluent network create --region`

References
----------

[CLI Commands Design](https://confluentinc.atlassian.net/wiki/spaces/PM/pages/2987394300/1-pager+--+Network+CLI)
[Implementation 1-Pager](https://confluentinc.atlassian.net/wiki/spaces/~712020463029eabbac401bbe1b033ce0d3a00e/pages/3136749631/Network+CLI)

Test & Review
-------------

Added integration tests.

Performed manual tests.

### CODE REVIEW

File: pkg/network/region.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	var regions []*region
+
+	for _, metadata := range metadataList {
+		if cloud != "" && cloud != metadata.GetId() {
+			continue
+		}
+
+		for _, r := range metadata.GetRegions() {
+			if len(r.Schedulability.DedicatedNetwork.DedicatedCluster.High) == 0 {
```
[the remark this answers is not in the document]
Comment (ay-zhang) [reply pkg/network/region.go:28]: Per capacity team, `is_schedulable` is a legacy flag and is basically not used. For our purposes the sub `schedulability` struct has the relevant info where `DedicatedCluster.High` indicates the region has [three or more schedulable zones](https://github.com/confluentinc/cc-scheduler-service/blob/354190c9fe02b47a4418273ce375b29d79a8671e/pg/mothership/service.go#L373) and `DedicatedCluster.Low` is less than three.

File: internal/network/command.go [hunk 2]
```diff
@@ -381,3 +383,24 @@ func (c *command) addNetworkLinkServiceFlag(cmd *cobra.Command) {
 	cmd.Flags().String("network-link-service", "", "Network link service ID.")
 	pcmd.RegisterFlagCompletionFunc(cmd, "network-link-service", c.validNetworkLinkServicesArgsMultiple)
 }
+
+func (c *command) addRegionFlagNetwork(cmd *cobra.Command, command *pcmd.AuthenticatedCLICommand) {
+	cmd.Flags().String("region", "", `Cloud region ID for this network.`)
```
Comment (brianstrauch) [inline internal/network/command.go:388]:
```suggestion
	cmd.Flags().String("region", "", "Cloud region ID for this network.")
```

File: internal/network/command_region.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package network
+
+import (
+	"github.com/spf13/cobra"
+)
+
+func (c *command) newRegionCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "region",
+		Short: "Manage Confluent Cloud regions.",
```
Comment (brianstrauch) [inline internal/network/command_region.go:10]:
```suggestion
		Short: "Manage Confluent Cloud network regions.",
```

File: internal/network/command_region_list.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/network"
+	"github.com/confluentinc/cli/v3/pkg/output"
+)
+
+func (c *command) newRegionListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List cloud provider regions.",
```
Comment (brianstrauch) [inline internal/network/command_region_list.go:14]:
```suggestion
		Short: "List cloud provider regions for networking.",
```

File: pkg/network/region.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ejagade) [reply pkg/network/region.go:28]: Sounds good. thanks for the code reference

**Review by ejagade — APPROVED** [review]

File: internal/network/command.go [hunk 5]
```diff
@@ -381,3 +383,24 @@ func (c *command) addNetworkLinkServiceFlag(cmd *cobra.Command) {
...
+
+		cloud, _ := cmd.Flags().GetString("cloud")
+		regions, err := network.ListRegions(command.Client, cloud)
+		if err != nil {
+			return nil
+		}
+
+		suggestions := make([]string, len(regions))
+		for i, region := range regions {
+			suggestions[i] = region.RegionId
```
Comment (brianstrauch) [inline internal/network/command.go:402]:
```suggestion
			suggestions[i] = region.GetRegionId()
```

Comment (brianstrauch) [reply internal/network/command.go:402]: Do we also want to include the region name here too?

**Review by brianstrauch — APPROVED** [review]

File: internal/network/command.go [hunk 6]
```diff
@@ -381,3 +383,24 @@ func (c *command) addNetworkLinkServiceFlag(cmd *cobra.Command) {
...
+
+		cloud, _ := cmd.Flags().GetString("cloud")
+		regions, err := network.ListRegions(command.Client, cloud)
+		if err != nil {
+			return nil
+		}
+
+		suggestions := make([]string, len(regions))
+		for i, region := range regions {
+			suggestions[i] = region.GetRegionId()
```
Comment (brianstrauch) [inline internal/network/command.go:402]:
```suggestion
			suggestions[i] = region.RegionId
```

**Pull request merged**